nbval/kernel.py

```diff
@@ -0,0 +1,112 @@
+"""
+pytest ipython plugin modification
+
+Authors: D. Cortes, O. Laslett, T. Kluyver, H. Fangohr, V.T. Fauske
+
+"""
+
+import os
+
+# Kernel for jupyter notebooks
+from jupyter_client.manager import KernelManager
+from jupyter_client.kernelspec import KernelSpecManager
+import ipykernel.kernelspec
+
+
+CURRENT_ENV_KERNEL_NAME = ':nbval-parent-env'
+
+
+class NbvalKernelspecManager(KernelSpecManager):
+    """Kernel manager that also allows for python kernel in parent environment
+    """
+
+    def get_kernel_spec(self, kernel_name):
+        """Returns a :class:`KernelSpec` instance for the given kernel_name.
+
+        Raises :exc:`NoSuchKernel` if the given kernel name is not found.
+        """
+        if kernel_name == CURRENT_ENV_KERNEL_NAME:
+            return self.kernel_spec_class(
+                resource_dir=ipykernel.kernelspec.RESOURCES,
+                **ipykernel.kernelspec.get_kernel_dict())
+        else:
+            return super().get_kernel_spec(kernel_name)
```

Comment:
This line should have been `super(NbvalKernelspecManager, self).get_kernel_spec(kernel_name)`, no?  It doesn't fail CI because the `--current-env` flag is always passed.